OrbitGl/PickingManager.cpp

```diff
@@ -8,52 +8,77 @@
 
 //-----------------------------------------------------------------------------
 PickingID PickingManager::CreatePickableId(Pickable* a_Pickable) {
+  absl::MutexLock lock(&mutex_);
   ++m_IdCounter;
-  PickingID id = PickingID::Get(PickingID::PICKABLE, ++m_IdCounter);
+  PickingID id = PickingID::Get(PickingID::PICKABLE, m_IdCounter);
   m_PickableIdMap[a_Pickable] = m_IdCounter;
   m_IdPickableMap[m_IdCounter] = a_Pickable;
   return id;
 }
 
 //-----------------------------------------------------------------------------
-void PickingManager::ClearIds() {
+void PickingManager::Reset() {
+  absl::MutexLock lock(&mutex_);
   m_IdPickableMap.clear();
   m_PickableIdMap.clear();
+  m_IdCounter = 0;
+}
+
+//-----------------------------------------------------------------------------
+Pickable* PickingManager::GetPickableFromId(uint32_t id) {
+  absl::MutexLock lock(&mutex_);
+  return m_IdPickableMap[id];
+}
+
+//-----------------------------------------------------------------------------
+Pickable* PickingManager::GetPicked() {
+  absl::MutexLock lock(&mutex_);
+  return m_Picked;
 }
 
 //-----------------------------------------------------------------------------
 void PickingManager::Pick(uint32_t a_Id, int a_X, int a_Y) {
-  m_Picked = m_IdPickableMap[a_Id];
-  if (m_Picked) {
-    m_Picked->OnPick(a_X, a_Y);
+  Pickable* picked = GetPickableFromId(a_Id);
+  if (picked) {
+    picked->OnPick(a_X, a_Y);
   }
+
+  absl::MutexLock lock(&mutex_);
```

Comment:
It looks like the lock is only added here because GetPickableFromId also locks. Maybe add a comment about that?

Reply:
It locks to protect access to m_Picked. GetPicked() also locks.